src/main/java/teammates/logic/automated/FeedbackSessionUnpublishedMailAction.java

```diff
@@ -0,0 +1,85 @@
+package teammates.logic.automated;
+
+import java.util.HashMap;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import teammates.common.datatransfer.FeedbackSessionAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+import teammates.common.util.Const.ParamsNames;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.HttpRequestHelper;
+import teammates.logic.core.EmailGenerator;
+import teammates.logic.core.FeedbackSessionsLogic;
+
+public class FeedbackSessionUnpublishedMailAction extends EmailAction {
+
+    private String feedbackSessionName;
+    private String courseId;
+    
+    public FeedbackSessionUnpublishedMailAction(HttpServletRequest request) {
+        super(request);
+        initializeNameAndDescription();
+        
+        feedbackSessionName = HttpRequestHelper
+                .getValueFromRequestParameterMap(request, ParamsNames.EMAIL_FEEDBACK);
+        Assumption.assertNotNull(feedbackSessionName);
+        
+        courseId = HttpRequestHelper
+                .getValueFromRequestParameterMap(request, ParamsNames.EMAIL_COURSE);
+        Assumption.assertNotNull(courseId);
+    }
+    
+    public FeedbackSessionUnpublishedMailAction(HashMap<String, String> paramMap) {
+        super();
+        initializeNameAndDescription();
+        
+        feedbackSessionName = paramMap.get(ParamsNames.EMAIL_FEEDBACK);
+        Assumption.assertNotNull(feedbackSessionName);
+        
+        courseId = paramMap.get(ParamsNames.EMAIL_COURSE);
+        Assumption.assertNotNull(courseId);
+    }
+    
+    @Override
+    protected void doPostProcessingForSuccesfulSend() throws InvalidParametersException, EntityDoesNotExistException {
+        FeedbackSessionAttributes feedbackObject = FeedbackSessionsLogic.inst()
+                .getFeedbackSession(feedbackSessionName, courseId);
+        feedbackObject.setPublishedEmailEnabled(false);
```

Comment:
I'll give this PR a closer look sometime later, but this looks a little odd. Should this be `setSentPublishedEmail` instead of `setPublishedEmailEnabled`?